From b03baef4b52483ac6ca7d9371224273b60455432 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Wed, 27 May 2020 14:10:33 +0100
Subject: [PATCH 4/7] build - systemd service files - install fix if systemd
 not found

so if systemd is not found - installe in PREFIX/lib/systemd/user which
allows build to work without disabling systemd.
---
 systemd-services/meson.build | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git systemd-services/meson.build systemd-services/meson.build
index 18be63c5ef..a17790e857 100644
--- systemd-services/meson.build
+++ systemd-services/meson.build
@@ -1,13 +1,16 @@
 if get_option('systemd') == true
-  dep = dependency('systemd')
+  sys_dep = dependency('systemd', required: false)
+  if sys_dep.found()
+    sys_dir = sys_dep.get_pkgconfig_variable('systemduserunitdir')
+  else
+    sys_dir = join_paths(dir_lib, 'systemd', 'user')
+  endif
 
   ethumb_service_config = configuration_data()
-
   ethumb_service_config.set('prefix', dir_prefix)
 
-  configure_file(
-    input: 'ethumb.service.in',
-    output: 'ethumb.service',
-    configuration: ethumb_service_config,
-    install_dir: dep.get_pkgconfig_variable('systemduserunitdir'))
+  configure_file(input         : 'ethumb.service.in',
+                 output        : 'ethumb.service',
+                 configuration : ethumb_service_config,
+                 install_dir   : sys_dir)
 endif
-- 
2.27.0

